api/src/main/java/org/pmiops/workbench/billing/BillingProjectBufferService.java

```diff
@@ -163,6 +162,32 @@ public void cleanBillingBuffer() {
             });
   }
 
+  private List<DbBillingProjectBufferEntry> findExpiredCreatingEntries(Instant now) {
+    return StreamSupport.stream(
+            billingProjectBufferEntryDao
+                .findAllByStatusAndLastStatusChangedTimeLessThan(
+                    DbStorageEnums.billingProjectBufferStatusToStorage(CREATING),
+                    new Timestamp(
+                        now.minus(CREATING_TIMEOUT_MINUTES, ChronoUnit.MINUTES).toEpochMilli()))
+                .spliterator(),
+            false)
+        // Ensure that we've also synchronized the status since the deadline elapsed. This
+        // covers degenerate cases where syncBillingProjectStatus might be backlogged and
+        // hasn't caught up to check all the CREATING projects, or covers periods where
+        // the Firecloud API is unresponsive. Ideally we'd push this logic directly down
+        // into the DAO, but JPQL has poor support for date comparisons. Keep the above
+        // filter as well as a first pass filter to limit what we pull into memory.
+        .filter(entry -> entry.getLastSyncRequestTime() != null)
```

Comment:
do we not also need to check `getLastStatusChangedTime()` for nullity?

Also, could you put a predicate method on `DbBillingProjectBufferEntry` that does this computation  as a derived property? Then it's just `.filter(DbBillingProjectBufferEntry::isInExpiredCreatingState)`. (May want to pass in the timestamp for testing though.)

Reply:
OK, I guess lots of folks prefer smart services and dumb entities.